### ISSUE TITLE
What is right way to disable eslint rules?

### ISSUE THREAD

**Issue by bondom** [eslint/typescript-eslint-parser#456]
**What version of TypeScript are you using?**
2.7.1

**What version of `typescript-eslint-parser` are you using?**
14.0.0

**What code were you trying to parse?**

```ts
// eslint-disable
const a = 3;
```

**What did you expect to happen?**
I expect to see clear code without warning

**What happened?**
I see warning: `[ts] 'a' is declared but its value is never read.`

But If I replace `// eslint-disable` with `@ts-ignore` everything works fine.

Is there way to disable rules with eslint-disable but not ts-ignore?

**Comment by mightyiam** [comment]
This?
https://eslint.org/docs/2.13.1/user-guide/configuring#disabling-rules-with-inline-comments

**Comment by bondom** [comment]
@mightyiam Sorry, but this doesn't work

**Comment by j-f1** [comment]
I think the error you’re encountering is a TypeScript error, and you have to use `// @ts-ignore` to silence it.

**Comment by mightyiam** [comment]
I recognize this line format: `[ts] 'a' is declared but its value is never read.` from VS Code. It seems like a TypeScript error, indeed. Otherwise, it would say `[eslint]...`, or so.

**Comment by bondom** [comment]
@j-f1 If this is TypeScript error, it should be controlled by compiler options in` tsconfig.json`, right?
There is my tsconfig.json:
```
{
  "compilerOptions": {
    "baseUrl": "./",
    "module": "esnext",
    "target": "es5",
    "lib": ["es6", "dom"],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "src",
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "allowUnusedLabels": true
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", ".tmp", "dist", "preprod"]
}

```

Also when I run `tsc --noEmit  --p ./tsconfig.json` I get no error

**Comment by bondom** [comment]
@mightyiam Yes! You are right! My TSLint extension was turned off

**Comment by bondom** [comment]
Thanks to all!
